controllers/serviceclaim_controller.go

```diff
@@ -488,7 +489,41 @@ func (r *ServiceClaimReconciler) DeleteServiceBindingsAndSecret(
 
 // SetupWithManager sets up the controller with the Manager.
 func (r *ServiceClaimReconciler) SetupWithManager(mgr ctrl.Manager) error {
+
+	genPred := predicate.GenerationChangedPredicate{}
+	reconcileOnRegisteredServiceUpdate := func(ctx context.Context, a client.Object) []reconcile.Request {
+		l := log.FromContext(ctx)
+		rs, ok := a.(*v1alpha1.RegisteredService)
+		if !ok {
+			l.Info("error parsing object to RegisteredService when mapping to ServiceClaim reconciliation trigger", "object", a)
+			return []reconcile.Request{}
+		}
+		if rs.Status.State != `Claimed` {
+			l.Info(fmt.Sprintf("Registered service %v is unclaimed, no need to reconcile service claim", rs.Name))
+			return []reconcile.Request{}
+		}
+		serviceclaims := &v1alpha1.ServiceClaimList{}
+		opts := &client.ListOptions{}
+		if err := r.List(ctx, serviceclaims, opts); err != nil {
+			l.Error(err,
+				"unable to list the ServiceClaims and reconcile for Registered Service Updates",
+				"RegisteredService", rs.Name)
+			return []reconcile.Request{}
+		}
+		reply := make([]reconcile.Request, 0, len(serviceclaims.Items))
```

Comment:
ServiceClaim - RegisteredService relationship is one to one. I don't think we need to look for more then one Service Claim. right?